builtin/logical/pki/path_tidy.go

```diff
@@ -778,6 +803,12 @@ func (b *backend) startTidyOperation(req *logical.Request, config *tidyConfig) {
 				return tidyCancelledError
 			}
 
+			if config.InvalidCerts {
+				if err := b.doTidyInvalidCerts(ctx, req, logger, config); err != nil {
```

Comment:
I don't know that I like the idea of this being a separate pass. We already have one pass which does this (`config.CertStore` controlling `b.doTidyCertStore`), so I think it is fine to reuse that for this pass as well. In the case of 1M+ certs, we don't really want to iterate over all the certificates twice, especially if most certs aren't invalid. :-)

My 2c!

Reply:
On line 784: `if config.CertStore || config.InvalidCerts {` ...

Reply:
Yes, I initially also wanted it to be a single pass but figured since the `tidy_invalid_certs` would be rarely used it wouldn't matter so much but this is much better, thank you :D